src/lib/is_val_object.js

```diff
@@ -0,0 +1,15 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+// returns true for a valid value object and false for tree nodes in the attribute hierarchy
+module.exports = function(obj) {
+    return obj && obj.valType !== undefined;
```

Comment:
Where do you specifically want to see it in `coerce.js`? Totally separate export (`exports.isValObject`), or as an element inside `exports.valObject`, e.g. `object: {...}` in which case the actual function is in `coerceFunction`, or something else?